Objects/genericaliasobject.c

```diff
@@ -525,12 +525,22 @@ _Py_subs_parameters(PyObject *self, PyObject *args, PyObject *parameters, PyObje
             return NULL;
         }
         if (unpack) {
+            if (!PyTuple_Check(arg)) {
+                Py_DECREF(newargs);
+                Py_DECREF(item);
+                Py_XDECREF(tuple_args);
+                PyErr_Format(PyExc_TypeError,
+                             "expected a tuple, not %T", arg);
```

Comment:
This error message is not helpful. Please add what method of what type has been called (this was method `__typing_subst__` of original argument `PyTuple_GET_ITEM(args, iarg)`).